File: 005/Problem5.mjs
import { performance } from "perf_hooks";

function problem() {
  var num = 3000;
  while (true) {
    if (num % 2 == 0) {
      if (num % 3 == 0) {
        if (num % 4 == 0) {
          if (num % 5 == 0) {
            if (num % 6 == 0) {
              if (num % 7 == 0) {
                if (num % 8 == 0) {
                  if (num % 9 == 0) {
                    if (num % 10 == 0) {
                      if (num % 11 == 0) {
                        if (num % 12 == 0) {
                          if (num % 13 == 0) {
                            if (num % 14 == 0) {
                              if (num % 15 == 0) {
                                if (num % 16 == 0) {
                                  if (num % 17 == 0) {
                                    if (num % 18 == 0) {
                                      if (num % 19 == 0) {
                                        if (num % 20 == 0) {
                                          break;
                                        }
                                      }
                                    }
                                  }
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
    num += 20;
  }
  console.log(num);
}

var start = performance.now();
for (var i = 0; i < 1000; i++) {
  problem();
}
var end = performance.now();
console.log("Function took " + (end - start) / 1000);

File: 004/Problem5.kt
import kotlin.system.measureNanoTime

fun main() {
    val loops = 1000
    val timeInNanos = measureNanoTime {
        for (i in 1..loops) {
            problem()
        }
    }
    println("Function Took $timeInNanos Nano seconds for $loops loops")
}

fun reverseInt(n: Int): Int {
    var num = n
    var newInt = 0
    var remainder: Int
    while (num != 0) {
        remainder = num % 10
        newInt *= 10
        newInt += remainder
        num /= 10
    }
    return newInt
}

fun problem() {
    var max = 0
    var prod: Int
    for (i in 100 until 1000) {
        for (j in 100 until 1000) {
            prod = i * j
            if (prod == reverseInt(prod) && prod > max) {
                max = prod
            }
        }
    }
    println(max)
}